### PR TITLE
Create Security Filter Sample

### PR DESCRIPTION
**Pull request by michaelScarfi**
This sample creates a Security filter given an session, filter expression, and a name for the new filter.

### CODE REVIEW

File: stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import com.microstrategy.webapi.EnumDSSXMLFolderNames;
+
+public class SecurityFilterCreation {
+
+	public static void main(String[] args) {	
+		// Connectivity for the intelligence server
+	    final String INTELLIGENCE_SERVER_NAME = "sup-w-001643.labs.microstrategy.com";
+	    final String PROJECT_NAME = "MicroStrategy Tutorial";
+	    final String MICROSTRATEGY_USERNAME = "Administrator";
+		final String MICROSTRATEGY_PASSWORD = "";
```
Comment (ddechent) [inline stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java:33]: Formatting

File: stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	 * Takes a WebIServerSession object, String expression, and String filterName and creates a Security Filter with the given expression and name
+	 * 
+	 * @param session Object of type WebIServerSession
+	 * @param expression String containing the expression for the Security Filter
+	 * @param filterName String containing the name for the new filter
+	 */
+	public static void createSecurityFilter(WebIServerSession session, String expression, String filterName){
+		// Create our Object Source, filter, and expression
+		WebObjectSource webObjectSource = session.getFactory().getObjectSource();
+	    WebMDSecurityFilter wMDFilter = (WebMDSecurityFilter) webObjectSource.getNewObject( EnumDSSXMLObjectTypes.DssXmlTypeMDSecurityFilter );
```
Comment (ddechent) [inline stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java:56]: Formatting

File: stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		WebExpression wExpr = wMDFilter.getExpression();
+		
+		try {
+			// Create folder object that will allow us to save the filter in the Security filter folder
+            String folderID = webObjectSource.getFolderID(EnumDSSXMLFolderNames.DssXmlFolderNameSystemMDSecurityFilters);
+            WebFolder folder = (WebFolder) webObjectSource.getObject(folderID, EnumDSSXMLObjectTypes.DssXmlTypeFolder);
```
Comment (ddechent) [inline stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java:62]: Formatting

File: stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	public static void main(String[] args) {	
+		// Connectivity for the intelligence server
+	    final String INTELLIGENCE_SERVER_NAME = "sup-w-001643.labs.microstrategy.com";
+	    final String PROJECT_NAME = "MicroStrategy Tutorial";
+	    final String MICROSTRATEGY_USERNAME = "Administrator";
+		final String MICROSTRATEGY_PASSWORD = "";
+		
+		// Filter information
+        String filterName = "test filter";
+		String expression = "[Employee Age]@ID=23";
```
Comment (ddechent) [inline stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java:37]: Formatting

File: stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+	    // Create our I-Server Session
+		WebIServerSession session = SessionManager.getSessionWithDetails(INTELLIGENCE_SERVER_NAME, PROJECT_NAME, MICROSTRATEGY_USERNAME, MICROSTRATEGY_PASSWORD);
```
Comment (ddechent) [inline stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java:40]: Formatting

File: stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+import com.microstrategy.web.objects.WebObjectsException;
+import com.microstrategy.web.objects.admin.users.WebMDSecurityFilter;
+import com.microstrategy.webapi.EnumDSSXMLObjectTypes;
+import com.microstrategy.webapi.EnumDSSXMLFolderNames;
+
+public class SecurityFilterCreation {
+
+	public static void main(String[] args) {	
+		// Connectivity for the intelligence server
+	    	final String INTELLIGENCE_SERVER_NAME = "sup-w-001643.labs.microstrategy.com";
```
Comment (Dholmblad) [inline stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java:30]: @michaelScarfi no need to include this server name in the sample.

File: stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+	 * Takes a WebIServerSession object, String expression, and String filterName and creates a Security Filter with the given expression and name
+	 * 
+	 * @param session Object of type WebIServerSession
+	 * @param expression String containing the expression for the Security Filter
+	 * @param filterName String containing the name for the new filter
+	 */
+	public static void createSecurityFilter(WebIServerSession session, String expression, String filterName){
+		// Create our Object Source, filter, and expression
+		WebObjectSource webObjectSource = session.getFactory().getObjectSource();
+	    	WebMDSecurityFilter wMDFilter = (WebMDSecurityFilter) webObjectSource.getNewObject( EnumDSSXMLObjectTypes.DssXmlTypeMDSecurityFilter );
```
Comment (Dholmblad) [inline stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java:56]: @michaelScarfi The space after parentasis is not necessary;

WebMDSecurityFilter webMdSecurityFilter = (WebMDSecurityFilter) webObjectSource.getNewObject(EnumDSSXMLObjectTypes.DssXmlTypeMDSecurityFilter);

File: stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+		WebExpression wExpr = wMDFilter.getExpression();
+		
+		try {
+			// Create folder object that will allow us to save the filter in the Security filter folder
```
Comment (Dholmblad) [inline stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java:60]: @michaelScarfi Might want to update the comment, this line is not creating a folder in the metadata. Its populating a java object with the system folder. Might be a bit confusing.

File: stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+	 * Takes a WebIServerSession object, String expression, and String filterName and creates a Security Filter with the given expression and name
+	 * 
+	 * @param session Object of type WebIServerSession
+	 * @param expression String containing the expression for the Security Filter
+	 * @param filterName String containing the name for the new filter
+	 */
+	public static void createSecurityFilter(WebIServerSession session, String expression, String filterName){
+		// Create our Object Source, filter, and expression
+		WebObjectSource webObjectSource = session.getFactory().getObjectSource();
+	    	WebMDSecurityFilter wMDFilter = (WebMDSecurityFilter) webObjectSource.getNewObject(EnumDSSXMLObjectTypes.DssXmlTypeMDSecurityFilter );
```
Comment (Dholmblad) [inline stand-alone/com/microstrategy/samples/objects/SecurityFilter/SecurityFilterCreation.java:56]: @michaelScarfi still have unnecessary whitespace in ' );'
